src/jump_wrapper.jl

```diff
@@ -0,0 +1,141 @@
+"""
+    diff_model(optimizer_constructor; with_parametric_opt_interface::Bool = true, with_bridge_type = Float64, with_cache::Bool = true)
+
+Create a JuMP model with a differentiable optimizer. The optimizer is created
+using `optimizer_constructor`. This model will try to select the proper
+differentiable optimization method based on the problem structure.
+
+See also: [`nonlinear_diff_model`](@ref), [`conic_diff_model`](@ref), [`quadratic_diff_model`](@ref).
+"""
+function diff_model(
+    optimizer_constructor;
+    with_parametric_opt_interface::Bool = true,
+    with_bridge_type = Float64,
+    with_cache::Bool = true,
+)
+    inner = diff_optimizer(
+        optimizer_constructor;
+        with_parametric_opt_interface = with_parametric_opt_interface,
+        with_bridge_type = with_bridge_type,
+        with_cache = with_cache,
+    )
+    return JuMP.direct_model(inner)
+end
+
+"""
+    nonlinear_diff_model(optimizer_constructor; with_bridge_type = Float64, with_cache::Bool = true)
+
+Create a JuMP model with a differentiable optimizer for nonlinear programs.
+The optimizer is created using `optimizer_constructor`.
+
+See also: [`conic_diff_model`](@ref), [`quadratic_diff_model`](@ref), [`diff_model`](@ref).
+"""
+function nonlinear_diff_model(
+    optimizer_constructor;
+    with_bridge_type = Float64,
+    with_cache::Bool = true,
+)
+    inner = diff_optimizer(
+        optimizer_constructor;
+        with_parametric_opt_interface = false,
```

Comment:
Add comment on why this is false

Reply:
Yup, I might re rewrite some of the internals to try to ger rid of this